edx_name_affirmation/models.py

```diff
@@ -45,6 +51,20 @@ class Meta:
         db_table = 'nameaffirmation_verifiedname'
         verbose_name = 'verified name'
 
+    @property
+    def verification_attempt_status(self):
+        "Returns the status associated with its SoftwareSecurePhotoVerification with verification_attempt_id if any."
+
+        if not self.verification_attempt_id or not SoftwareSecurePhotoVerification:
+            return None
+
+        try:
+            verification = SoftwareSecurePhotoVerification.objects.get(id=self.verification_attempt_id)
+            return verification.status if verification else None
```

Comment:
Nit: I think that if `verification` were to be `None`, theoretically, `get` would have raised the `SoftwareSecurePhotoVerification.DoesNotExist` exception instead. I don't think there's a case where `verification` could be `None`. I think that the `None`-checking is not necessary.

Reply:
Correct. The check is not necessary. I'll remove it.